presto-base-jdbc/src/main/java/io/prestosql/plugin/jdbc/JdbcConnectorFactory.java

```diff
@@ -83,14 +88,14 @@ public Connector create(String catalogName, Map<String, String> requiredConfig,
         return injector.getInstance(JdbcConnector.class);
     }
 
+    @Deprecated
     public interface JdbcModuleProvider
     {
         Module getModule(String catalogName);
+    }
 
-        static JdbcModuleProvider withCredentialProvider(Module module)
-        {
-            requireNonNull(module, "module is null");
-            return catalogName -> combine(new CredentialProviderModule(), module);
-        }
+    public static Module withCredentialProvider(Module module)
```

Comment:
You're not maintaining compatibility (the FQN name of the method changed), so you can as well remove this method altogether (inline). I think we should rather pull `CredentialProviderModule` explicitly.